File: 234. Palindrome_Linked_List/solution.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    bool isPalindrome(ListNode* head) {
        
        stack<int> st;
        
        ListNode* it = head;
        
        while(it != nullptr){
            st.push(it->val);
            it = it->next;
        }
        
        it = head;
        
        while(it != nullptr){
            if(st.top() == it->val)
                st.pop();
            else
                break;
            it = it->next;
        }
        
        return st.size() == 0 ? true : false;
        
    }
};

File: 234. Palindrome_Linked_List/solution.java

/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public boolean isPalindrome(ListNode head) {
        
        Stack<Integer> st = new Stack<Integer>();
        
        ListNode it = head;
        
        while(!Objects.isNull(it)){
            st.push(it.val);
            if(!Objects.isNull(it.next))
                it = it.next;
            else
                break;
        }
        
        it = head;
        
        while(!Objects.isNull(it)){
            if(it.val == st.peek())
                st.pop();
            else
                break;
            if(!Objects.isNull(it.next))
                it = it.next;
            else
                break;
        }
        
        return st.empty();
        
    }
}